app/components/leaflet-map.js

```diff
@@ -203,6 +215,30 @@ export default Ember.Component.extend({
     if (!this.get('geoJSON')) {
       this.map.drawnItems.clearLayers();
     }
-  })
+  }),
+
+  //ids for cities, their locations and zoom.
```

Comment:
I was expecting to see the concept of a city only live in the parent component. 

I think some good inspiration on separating our concerns can come from the [Ember Leaflet](http://www.ember-leaflet.com/docs/actions) project.
```hbs
<p>Latitude: <code>{{lat}}</code> / Longitude: <code>{{lng}}</code></p>
{{#leaflet-map lat=lat lng=lng zoom=zoom onMoveend=(action "updateCenter")}}
  {{tile-layer url="http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png"}}
{{/leaflet-map}}
```

Taking an approach like that and making lat and lon attributes that we pass to the map will make the map more flexible and let us isolate everything about cities in one place. 

You could also consider making a new component that wraps leaflet-map and adds the city hopping (similarly to how the heat-map component wraps leaflet-map). But that would probably make your nice symmetric design with the drop-down on the left unworkable. Just spitballing